packages/utxo-lib/src/compose/transaction.ts

```diff
@@ -5,76 +5,30 @@ import {
     ComposeChangeAddress,
     ComposeFinalOutput,
     ComposedTransaction,
-    CoinSelectOutputFinal,
+    TransactionInputOutputSortingStrategy,
 } from '../types';
-
-function convertOutput(
-    selectedOutput: CoinSelectOutputFinal,
-    composeOutput: ComposeFinalOutput | ({ type: 'change' } & ComposeChangeAddress),
-) {
-    if (composeOutput.type === 'change') {
-        return {
-            ...composeOutput,
-            amount: selectedOutput.value.toString(),
-        };
-    }
-
-    if (composeOutput.type === 'opreturn') {
-        return composeOutput;
-    }
-
-    return {
-        ...composeOutput,
-        type: 'payment' as const,
-        amount: selectedOutput.value.toString(),
-    };
-}
-
-function inputComparator(a: ComposeInput, b: ComposeInput) {
-    return Buffer.from(a.txid, 'hex').compare(Buffer.from(b.txid, 'hex')) || a.vout - b.vout;
-}
-
-function outputComparator(a: CoinSelectOutputFinal, b: CoinSelectOutputFinal) {
-    return (
-        a.value.cmp(b.value) ||
-        (Buffer.isBuffer(a.script) && Buffer.isBuffer(b.script)
-            ? a.script.compare(b.script)
-            : a.script.length - b.script.length)
-    );
-}
+import { noneSortingStrategy } from './sorting/noneSortingStrategy';
+import { SortingStrategy } from './sorting/sortingStrategy';
+import { bip69SortingStrategy } from './sorting/bip69SortingStrategy';
 
 export function createTransaction<Input extends ComposeInput, Change extends ComposeChangeAddress>(
     request: ComposeRequest<Input, ComposeFinalOutput, Change>,
     result: CoinSelectSuccess,
 ): ComposedTransaction<Input, ComposeFinalOutput, Change> {
-    const convertedInputs = result.inputs.map(input => request.utxos[input.i]);
-
-    const defaultPermutation: number[] = [];
-    const convertedOutputs = result.outputs.map((output, index) => {
-        defaultPermutation.push(index);
-        if (request.outputs[index]) {
-            return convertOutput(output, request.outputs[index]);
-        }
-
-        return convertOutput(output, { type: 'change', ...request.changeAddress });
-    });
+    const sortingStrategy: TransactionInputOutputSortingStrategy =
+        // eslint-disable-next-line no-nested-ternary
+        request.sortingStrategy === undefined
+            ? request.skipPermutation === true
+                ? 'none'
+                : 'bip69'
+            : request.sortingStrategy;
```

Comment:
Nit: what do you think about this instead of nested ternary? 
```suggestion
    const sortingStrategy: TransactionInputOutputSortingStrategy =
        request.sortingStrategy ?? (request.skipPermutation === true ? 'none' : 'bip69');
```

Reply:
EDIT: outdated, feel free to ignore and resolve, the new function is good :ok_hand:

Reply:
:+1: resolving